components/aws/sagemaker/tests/integration_tests/utils/kfp_client_utils.py

```diff
@@ -23,9 +24,20 @@ def compile_and_run_pipeline(
     return run.id
 
 
-def wait_for_job_completion(client, run_id, timeout):
+def wait_for_job_completion(client, run_id, timeout, status_to_check):
     response = client.wait_for_run_completion(run_id, timeout)
-    status = response.run.status.lower() == "succeeded"
+    status = response.run.status.lower() == status_to_check
+    return status
+
+
+def wait_for_job_status(client, run_id, timeout, status_to_check="succeeded"):
+    if status_to_check == "succeeded":
+        status = wait_for_job_completion(client, run_id, timeout, status_to_check)
+    else:
+        time.sleep(timeout)
+        response = client.get_run(run_id)
```

Comment:
clarification needed: do we need to do this ?

I think the `wait_for_run_completion()` method does not error out even if job is not finished by the timeout time, does it already return its status as running after timeout `response.run.status.lower() == status_to_check` might be sufficient ?

Reply:
True, that might be sufficient but IMO this is ok too. Is there any specific reason why you prefer that ?